lib/http_basic_auth.rb

```diff
@@ -0,0 +1,16 @@
+# Encapuslates the parameters to use for basic HTTP authentication
+#
+# @see https://docs.ruby-lang.org/en/2.7.0/Net/HTTP.html#class-Net::HTTP-label-Basic+Authentication
+class HTTPBasicAuth
+  attr_reader :user
+  attr_reader :password
+
+  # A new instance of +HTTPBasicAuth+
+  #
+  # @param user [String] the username
+  # @param password [String] the password
+  def initialize(user:, password:)
+    @user = user
+    @password = password
+  end
+end
```

Comment:
Tip: You could also just use ruby's `Struct` to create data classes like that in a more concise way (could be useful too for when you decide to convert your `projects` into one too) with a oneliner like that:

```
HTTPBasicAuth = Struct.new(:user, :password, keyword_init: true)
```

See https://www.rubyguides.com/2017/06/ruby-struct-and-openstruct/

Not sure if you find it more readable or more obscure… 🤷